sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -88,6 +88,7 @@ private[csv] class CSVOptions(
 
   val quote = getChar("quote", '\"')
   val escape = getChar("escape", '\\')
+  val escapeEscape = getChar("escapeEscape", quote)
```

Comment:
Also, I would make this as an option for this CSV datasource. I think we should document this in `DataFrameReader`, `DataStreamReader`, `readwriter.py` and `streaming.py`. 

Lastly, how about making this from `escapeEscape ` to `escapeQuoteEscaping` in order to take after the original name.